unit-tests/src/main/scala/java/net/URISuite.scala

```diff
@@ -0,0 +1,169 @@
+  }
+
+  test("should parse relative URIs") {
+    expectURI(new URI("docs/guide/collections/designfaq.html#28"),
+              false,
+              false)(path = "docs/guide/collections/designfaq.html",
+                     fragment = "28",
+                     schemeSpecificPart =
+                       "docs/guide/collections/designfaq.html")()
+    expectURI(new URI("../../../demo/jfc/SwingSet2/src/SwingSet2.java"),
+              false,
+              false)(path = "../../../demo/jfc/SwingSet2/src/SwingSet2.java",
+                     schemeSpecificPart =
+                       "../../../demo/jfc/SwingSet2/src/SwingSet2.java")()
+  }
+
+  test("should fail on bad URIs") {
+    val badURIs = Array(
+      "http:///a path#frag", // space char in path, not in escaped
+      // octet form, with no host
+      "http://host/a[path#frag", // an illegal char, not in escaped
+      // octet form, should throw an
+      // exception
+      "http://host/a%path#frag", // invalid escape sequence in path
+      "http://host/a%#frag", // incomplete escape sequence in path
+      "http://host#a frag",  // space char in fragment, not in
+      // escaped octet form, no path
+      "http://host/a#fr#ag", // illegal char in fragment
+      "http:///path#fr%ag",  // invalid escape sequence in fragment,
+      // with no host
+      "http://host/path#frag%", // incomplete escape sequence in
+      // fragment
+      "http://host/path?a query#frag", // space char in query, not
+      // in escaped octet form
+      "http://host?query%ag", // invalid escape sequence in query, no
+      // path
+      "http:///path?query%", // incomplete escape sequence in query,
+      // with no host
+      "mailto:user^name@fklkf.com", // invalid char in scheme specific part
+      // authority validation
+      "http://user@[3ffe:2x00:100:7031::1]:80/test", // malformed
+      // IPv6 authority
+      "http://[ipv6address]/apath#frag", // malformed ipv6 address
+      "http://[ipv6address/apath#frag", // malformed ipv6 address
+      "http://ipv6address]/apath#frag", // illegal char in host name
+      "http://ipv6[address/apath#frag",
+      "http://ipv6addr]ess/apath#frag",
+      "http://ipv6address[]/apath#frag",
+      // illegal char in username...
+      "http://us[]er@host/path?query#frag",
+      "http://host name/path", // illegal
+      // char in authority
+      "http://host^name#fragment", // illegal char in authority
+      "telnet://us er@hostname/",  // illegal char in authority
+      // missing components
+      "//", // Authority expected
+      "ascheme://", // Authority expected
+      "ascheme:",   // Scheme-specific part expected
+      // scheme validation
+      "a scheme://reg/", // illegal char
+      "1scheme://reg/", // non alpha char as 1st char
+      "asche\u00dfme:ssp", // unicode char , not USASCII
+      "asc%20heme:ssp"
+    )
+
+    for (uri <- badURIs) {
+      assertThrows[URISyntaxException] { new URI(uri) }
+    }
+  }
+
```

Comment:
Are there tests in Apache Harmony too? Why not port them? Why are you porting `java.net.URI` from Apache Harmony and the tests from Scala.js? Why not port `java.net.URI` from Scala.js?